In [15]:
from pymatgen import Structure
from pymatgen.ext.matproj import MPRester
with MPRester("ZPSKgimujRNFyTRi") as m:

    # Structure for material id
    structure = m.get_structure_by_material_id("mp-961687")

In [9]:
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer

#Création d'un objet SpaceGroupAnalyzer
struct = SpacegroupAnalyzer(structure)
new_s=SpacegroupAnalyzer(structure).get_primitive_standard_structure()

In [10]:
#Obtenir les vecteurs des réseaux direct et réciproque
#1 réseau direct :
resd = new_s.lattice
a = resd.a
b = resd.b
c = resd.c
#2 réseau réciproque :
resrec = resd.reciprocal_lattice
aa = resrec.a
bb = resrec.b
cc = resrec.c
print (a,b,c,aa,bb,cc)

4.529291876717375 4.529291876717375 4.529291876717376 1.6990070833210216 1.6990070833210218 1.6990070833210218


# Vecteurs
Les vecteurs de bases des ses réseaux direct et réciproque sont donnés par les paramètres a,b,c et a*,b*,c* respectivement :
***
* Réseau direct :
a = 4.529291876717375 angström 
b = 4.529291876717375 angström
c = 4.529291876717376 angström
* Réseau réciproque :
a* = 1.6990070833210216 angström
b* = 1.6990070833210218 angström
c* = 1.6990070833210218 angström

In [12]:
#Type de maille
maille = struct.get_lattice_type()
print(maille)

cubic


# Type de maille
Le type de maille est déterminé à l'aide de la fonction get_lattice_type(). 
***
La maille de ce cristal est cubique.

In [13]:
#Système Crtistallin 
système = struct.get_crystal_system()
print(système)

cubic


# Système Cristallin 
Le système est déterminé à l'aide de la fonction get_crystal_system().
***
Le système est cubique

In [16]:
#Groupe ponctuel
GP = struct.get_point_group_symbol()
print(GP)

-43m


# Groupe Ponctuel
Le groupe ponctuel est déterminé grâce à la fonction get_point_group_symbol()
***
Le groupe ponctuel de ce cristal est : F -4 3 m
